fendermint/vm/interpreter/src/fvm/gas/eip1559.rs

```diff
@@ -0,0 +1,108 @@
+use crate::fvm::gas::{Gas, GasMarket};
+use crate::fvm::state::{read_actor_state, FvmExecState};
+use crate::fvm::FvmMessage;
+use anyhow::anyhow;
+use fendermint_actor_gas_market::EIP1559GasState;
+use fendermint_vm_actor_interface::gas::GAS_ACTOR_ADDR;
+use fendermint_vm_actor_interface::system;
+use fvm_ipld_blockstore::Blockstore;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+type AtomicGas = AtomicU64;
+
+/// The gas market based on EIP1155
+#[derive(Default)]
+pub struct EIP1559GasMarket {
+    /// The block gas limit
+    block_gas_limit: AtomicGas,
+    /// The accumulated gas usage so far
+    block_gas_used: AtomicGas,
```

Comment:
Is `AtomicGas` needed purely due to the ownership issues you pointed out on in our last sync? If so, please add a note here + a TODO, since this component is single-threaded and we should attempt to remove this overhead.

Reply:
Purely because of the trait bound, I will add the note and todo